k8s/telemetry.go

```diff
@@ -0,0 +1,24 @@
+package k8s
+
+import (
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+)
+
+var tracer trace.Tracer
+
+// SetTracer sets the tracer used for generating spans for this package
+func SetTracer(t trace.Tracer) {
+	tracer = t
```

Comment:
NIT – these can cause data races. I don't think the probability is high, but the impact is a panic, so maybe worth adding an rwmutex here.